packages/rocketchat-lib/client/lib/openRoom.js

```diff
@@ -9,7 +9,7 @@ function openRoom(type, name) {
 	return Meteor.defer(() =>
 		currentTracker = Tracker.autorun(function(c) {
 			const user = Meteor.user();
-			if ((user && user.username == null) || user == null && RocketChat.settings.get('Accounts_AllowAnonymousRead') === false) {
+			if ((user == null || user.username == null) && RocketChat.settings.get('Accounts_AllowAnonymousRead') === false) {
```

Comment:
I'm not sure about this change, It will allow users without username on servers with anonymous read true to open a room instead of show the screen to select usernames right?

Reply:
ok.. the logic change was unintended.. fixing